20_2 DataGrid Group Summaries: rewrite (#617)
DevExpress/devextreme-demos: JSDemos/Demos/DataGrid/GroupSummaries/description.md

> To configure group summaries, populate the [summary](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/).[groupItems](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/) array with summary configuration objects. Each object should specify a [column](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#column) that supplies data for summary calculation and a [summaryType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#summaryType) &mdash; an aggregate function that should be applied to this data.

Reviewer: ```suggestion
To configure group summaries, populate the [summary](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/).[groupItems](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/) array with summary configuration objects. Each object should specify a [column](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#column) that supplies data for summary calculation and a [summaryType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#summaryType)&mdash;an aggregate function that should be applied to this data.
```

### Position a Group Summary

Group summaries are displayed in parentheses after the group caption. You can reposition them as follows:

* Display a summary in a column          

> You can align a summary with the column by which it was calculated (see the Sale Amount and Total Amount columns). To do this, enable the [alignByColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#alignByColumn) property. If you want to align a summary with a column of your choice, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.

Reviewer 2: Reads like a complicated phrase: `align a summary with the column by which it was calculated`. 

```suggestion
You can align a summary to its corresponding column (see the Sale Amount and Total Amount columns). To do this, enable the [alignByColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#alignByColumn) property. If you want to align a summary to a different column, specify the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.
```

* Move a summary to the group footer       

> To position a summary in the group footer, enable the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property. The summary will be aligned with the column by which it was calculated (see the Total Amount column). If you want to use another column, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.

Reviewer 2: `a summary in the group footer` - we're talking about one summary and "the" group footer. Let's rephase it to indicate that the grid displays multiple summary values in multiple group rows/footers. 

Trying to also get rid of passive in the second sentence. 

```suggestion
To display summary values in group footers, enable the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property. Summary items will appear within their corresponding column (see the Total Amount column). If you want to display summaries in another column, specify the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.
```

Author: @Abadzhev 
This paragraph explains how to move one summary to the group footer. Not all summaries. That's why singular fits better here. Can this be applied:
"To display a summary value in the group footer, enable the [showInGroupFooter]() property. The Summary will appear in its corresponding column (see the Total Amount column). If you want to use another column, specify it in the [showInColumn] property."
?

Reviewer 2: So let's enumerate the terms that we use: 

- Summary, Total Summary, Group Summary: this is a generic term, a feature name. You use it throughout the topic in the sense of "summary item", or "summary value". Sometimes it works fine. "You can add a group summary to the grid", "The grid supports group summaries", "The footer displays a total summary" - these are fine as synonyms to summary item, or summary value, or as a feature name. 
- Summary item: an object that defines calculation settings. It can produce one value (total summary), or multiple values (group summary). 
- Summary value: the result of summary calculation (as mentioned above, there can be multiple values per summary item). This is what the grid actually displays - calculation results, values. 

Now let's return to the sentence in question, and start with "group footer". You can't use "the group footer", because this phrase implies that it's one and only footer that we know about. But there are multiple group rows (unless there's only one group). You can say "the grid's footer" - that one in fact is specific - only one instance exists per grid / view.

So we need to use plural for "group footer". Then if we do that, then "group footers display summary values". In other words: one summary item produced multiple values and the grid displays those values in group footers. 

If you say "a summary value", you mean one value, that is a single group footer cell.  

I agree my version is not explicit about values being all produced by a single summary item. (I don't think it's necessary to specify, since we talk about summary item settings...) But we can be specific, if you want: 

"You can make a summary item display its values within group footers. Set the ..."  

I hope that you can find a version that works based on these comments.

Author: How about substituting "the group footer" for "the footer of its group" ():
"Move a summary to the footer of its group
To position a summary in the footer of its group, enable the showInGroupFooter property. The summary will be aligned to its corresponding column (see the Total Amount column). If you want to use another column, specify it in the showInColumn property."
?

Reviewer 2: It still leaves the definite article, and the meaning is pretty much the same: I don't see much difference between "the group footer" or "the footer of its group". The latter seems a bit more complicated and confusing. I don't know why you are fighting the plural so much :-) A summary item does display values in multiple group footers. I think it's all technically correct and simple enough... 

Display Summary Values within Group Footers
You can make a summary item display its values within group footers. To do that, set ...

> You can make a summary item display its values within group footers. To do that, enable the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property. Summary items will appear within their corresponding columns (see the Total Amount column). If you want to display summaries in another column, specify the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.

Reviewer 3: ```suggestion
You can make a summary item display its values within group footers. To do this, enable the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property. Summary items will appear within their corresponding columns (see the Total Amount column). If you want to display summaries in another column, specify the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.
```

### Format Summary Values

> You can specify a text for the summary value. For this, specify it in the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property (see the summaries of type *"sum"* and *"count"*). You can also set a custom display format for a summary item value in the [valueFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#valueFormat) property (see the summaries for the Total Amount and Sale Amount columns).

Reviewer: It's the other way around:

```suggestion
You can format a summary value in the [valueFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#valueFormat) property. This demo specifies the *"currency"* **valueFormat** for the Total Amount and Sale Amount summary values. You can also use the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property to set a custom display format for summary values. For example, you can add custom text to a summary value, as shown in this demo for summaries of the *"sum"* and *"count"* types.
```

> To format summary values, use the [valueFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#valueFormat) property. This demo applies the *"currency"* format to summary values calculated against the Total Amount and Sale Amount columns. If you also want to add custom text to summary values, use the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property. This demo sets this property for summaries of *"sum"* and *"count"* types.

Reviewer 3: demo?

Author: Yes, demo - it is available here: https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/GroupSummaries/React/Light/

Reviewer 3: ok thanks

> You can format a summary value in the [valueFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#valueFormat) property. This demo specifies the *"currency"* **valueFormat** for the Total Amount and Sale Amount summary values. You can also use the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property to set a custom display format for summary values. For example, you can add custom text to a summary value, as shown in this demo for summaries of the *"sum"* and *"count"* types.

Reviewer 2: Format something "IN" the property - this preposition doesn't work. 

You also say "a summary value", but we need to refer to multiple values, since we're talking about group summaries. 

`Total Amount and Sale Amount summary values` - this phrase is a bit misleading. Those a column names, so it's the format of summary values calculated against those columns. 

You don't explain what is "a custom display format". You give one example, which is "add text". But what else? If it's only that (in addition to actually formatting the value), then I give a suggestion below. 

```suggestion
To format summary values, use the [valueFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#valueFormat) property. This demo applies the *"currency"* format to summary values calculated against the Total Amount and Sale Amount columns. If you also want to add custom text to summary values, use the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property. This demo sets this property for summaries of *"sum"* and *"count"* types.
```

### Sort Groups by Summary Values

> The DataGrid sorts groups by their captions. To sort them by summary values, populate the [sortByGroupSummaryInfo](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/) array with objects. Each object should include the [summaryItem](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/#summaryItem) property that specifies a group summary item by which to sort groups. In this demo, the **sortByGroupSummaryInfo** array contains a single object that sorts groups by the number of orders.

Reviewer 2: `The DataGrid sorts groups by their captions` - a "caption" is the wrong word to use here. Also, this first sentence describes the "default" state of things. 

We could say something like this: 

"The grid always sorts data groups. The default sort order depends on the corresponding group column values. For example, if you group data by Country, then groups will be sorted alphabetically by Country names. 

You may also want to sort group rows by their corresponding summary values. For example, you can sort countries by the number of fulfilled orders. To do that, use the ... "

But in the end, that's probably too basic. Then why don't we compress the text to the following fragment:

```suggestion
This demo sorts groups by summary values (the number of orders). To do this, the code adds the corresponding summary item to the [sortByGroupSummaryInfo](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/) array. 
```

Author: @Abadzhev 
The suggested text doesn't mention that the sortByGroupSummaryInfo array must consist of objects and that each object must specify the summaryItem property.
Also, the code doesn't "add" anything - it's the user who does that. But let's see what the final edit is.
Can the paragraph be
"This demo sorts groups by summary values (the number of orders). To do this, add an object to the [sortByGroupSummaryInfo]() array and specify the summary of type "count" in the summaryItem property."?

Reviewer 2: Maybe I misunderstood something, but I disagree about "add". User (I assume you mean developer) writes code. Code (a line in that code, a command) adds an item to the array. A user cannot add an item to an array. I can say that figuratively, but it doesn't mean that "the code does something" is incorrect. If you have a code sample, you can always say "the following code does X". 

In your suggestion, you say "To do this, add". As I understand, the demo already does this. We need to instruct the customer to look into code, not to write code. 

Here's another suggestion then...

```suggestion
This demo sorts groups by summary values (the number of orders). To do this, the code adds an object to the [sortByGroupSummaryInfo](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/) array. The object specifies the [summaryItem](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/#summaryItem) whose values are used to sort the group rows. 
```

> A group summary creates a synopsis for records in a group. To configure this feature, populate the [summary](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/).[groupItems](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/) array with summary configuration objects. Each object should specify a [column](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#column) that supplies data for summary calculation and a [summaryType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#summaryType) &mdash; an aggregate function that should be applied to this data.

Reviewer: Em dash doesn't need spaces.

```suggestion
A group summary creates a synopsis for records in a group. To configure this feature, populate the [summary](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/).[groupItems](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/) array with summary configuration objects. Each object should specify a [column](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#column) that supplies data for summary calculation and a [summaryType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#summaryType)&mdash;an aggregate function that should be applied to this data.
```

### Position a Group Summary

> The default position for group summaries are the parentheses after the group caption. You can change the position to one of the following:

Reviewer: ```suggestion
Group summaries are displayed in parentheses after the group caption. You can reposition them as follows:
```

> DataGrid will use the value from the column property of the summary. 

Reviewer: I suppose this sentence was forgotten to be removed.

```suggestion
```

> * Group row's column

Reviewer: ```suggestion
* Display a summary in a column          
```

> To position a summary in the group row, set the [alignByColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#alignByColumn) property to **true**. DataGrid will position the summary in the column specified in the summary's **column** property. If you want to use another column, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.

Reviewer: ```suggestion
You can align a summary with the column by which it was calculated (see the Sale Amount and Total Amount columns). To do this, enable the [alignByColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#alignByColumn) property. If you want to align a summary with a column of your choice, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.
```

> * Group footer

Reviewer: ```suggestion
* Move a summary to the group footer       
```

> To position a summary in the group footer, set the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property to **true**. DataGrid aligns summaries to the right of the footer. If you want to display a summary in a certain column of the footer, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.

Reviewer: ```suggestion
To position a summary in the group footer, enable the [showInGroupFooter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInGroupFooter) property. The summary will be aligned with the column by which it was calculated (see the Total Amount column). If you want to use another column, specify it in the [showInColumn](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#showInColumn) property.
```

### Format Summary Values

> To set a text for the summary value, specify it in the [displayFormat](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/groupItems/#displayFormat) property.

Reviewer: What about the `valueFormat`?
Don't forget to make references to the demo itself so that the reader knows where to look in the code.

### Sort Groups by Summary Values

> To define a display order for groups, populate the [sortByGroupSummaryInfo](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/) array with objects. Each object must consist of at least the [summaryItem](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/#summaryItem) property that specifies the group summary item by which to sort groups. In this demo, groups are sorted be the number of orders.

Reviewer: ```suggestion
The DataGrid sorts groups by their captions. To sort them by summary values, populate the [sortByGroupSummaryInfo](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/) array with objects. Each object should include the [summaryItem](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/sortByGroupSummaryInfo/#summaryItem) property that specifies a group summary item by which to sort groups. In this demo, the **sortByGroupSummaryInfo** array contains a single object that sorts groups by the number of orders.
```
